### ISSUE TITLE
Occasion 'nan' in training w/. semihard triplet loss

### ISSUE THREAD

**Issue by SuperbTUM** [alfonmedela/TripletSemiHardLoss-PyTorch#3]
Hi, thanks for your work. I tried with your semihard triplet loss and found it occasionally becomes nan in training. I tried to add a trivial bias [here](https://github.com/alfonmedela/triplet-loss-pytorch/blob/master/loss_functions/triplet_loss.py#L105) for robustness but it still did not work as expected. Could you please help me review your code again to see if there is anything we are missing? Thanks!

**Comment by alfonmedela** [comment]
Hi 😄

How many categories do you have in the training set? It is easier to have NAN values in the validation set rather than in the training set due to missing similar pairs. 

The easiest way to solve this is to increase the batch size but, of course, this will increase the computational cost. This might be a solution if you are using a very small batch size or if your number of categories is not very large. A batch size of 128 and 256 usually works with ~1,000 categories or more.

**Comment by SuperbTUM** [comment]
Understood, thanks. I compared your code with TensorFlow addons as well and your implementation should be good.

**Comment by alfonmedela** [comment]
Happy to help
